packages/cashier/src/containers/cashier/cashier.jsx

```diff
@@ -67,12 +68,14 @@ const Cashier = ({
         const options = [];
         routes_config.forEach(route => {
             if (
-                !route.is_invisible &&
-                (route.path !== routes.cashier_pa || is_payment_agent_visible) &&
-                (route.path !== routes.cashier_pa_transfer || is_payment_agent_transfer_visible) &&
-                (route.path !== routes.cashier_p2p || is_p2p_enabled) &&
-                (route.path !== routes.cashier_onramp || is_onramp_tab_visible) &&
-                (route.path !== routes.cashier_acc_transfer || is_account_transfer_visible)
+                (is_cra && !route.hide_for.includes('affiliate')) ||
+                (!is_cra &&
+                    !route.is_invisible &&
+                    (route.path !== routes.cashier_pa || is_payment_agent_visible) &&
+                    (route.path !== routes.cashier_pa_transfer || is_payment_agent_transfer_visible) &&
+                    (route.path !== routes.cashier_p2p || is_p2p_enabled) &&
+                    (route.path !== routes.cashier_onramp || is_onramp_tab_visible) &&
+                    (route.path !== routes.cashier_acc_transfer || is_account_transfer_visible))
```

Comment:
It's a big condition, Would be nice if we could extract it into a method.

Reply:
@farzin-fs sorted